src/app/features/game/game.component.ts

```diff
@@ -1,15 +1,88 @@
-import { Component } from '@angular/core';
+import {Component, OnInit} from '@angular/core';
+
+function getRandomInt(min: number, max: number) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
 
 @Component({
   selector: 'idle-game',
   standalone: true,
   templateUrl: './game.component.html',
   styleUrl: './game.component.scss'
 })
-export class GameComponent {
+export class GameComponent implements OnInit {
   count = 0;
 
-  incrementCount() {
-    this.count++;
+  intervalId: number | null = null;
+
+  clickGadgets = {
+    doubleTap: {
+      num: 0,
+      cost: 10
+    },
+    autoClicker: {
+      num: 0,
+      cost: 100
+    },
+    luckyFinger: {
+      num: 0,
+      cost: 1000
+    }
+  };
+
+  get gadgetNum() {
+    return this.clickGadgets.doubleTap.num + this.clickGadgets.autoClicker.num + this.clickGadgets.luckyFinger.num;
+  }
+
+  clickButton() {
+    this.count += getRandomInt(1, 1 + this.clickGadgets.luckyFinger.num);
+  }
+
+  autoClickButton() {
+    this.clickButton();
+  }
+
+  manualClickButton() {
+    this.clickButton();
+    this.count += this.clickGadgets.doubleTap.num;
+  }
+
+  ngOnInit() {
+    this.adjustInterval();
+  }
+
+  buyDoubleTapGadget() {
```

Comment:
This is fine for now, but i think in the future, we can have a file that just has all the buy options and we can use an ngFor loop in order to create the options in the HTML (this will automatically add any new objects). We can change the click functions that take a single arg of the name of the item and updates the variables accordingly so we don't have to add a function, a value, an HTML button, etc. everytime we want to add an option.